ZenPacks/zenoss/ZenPackLib/lib/base/ComponentBase.py

```diff
@@ -231,6 +231,13 @@ def get_containing_relname(self):
         """Return name of containing relationship."""
         for relname, relschema in self._relations:
             if issubclass(relschema.remoteType, ToManyCont):
+                # skip if it's not the real containing rel
+                # for instance an inherited os.filesystems for sub-classed FileSystem
+                rel = getattr(self, relname, None)
```

Comment:
This gets into the fact that ZPL by default roots component relations to the device, not the os or hw subcomponents of the device.  This created a problem for the ZPL Proxy subclasses of the traditional classes like FileSystem or OsComponent. The ZPL suclass roots these at device.filesystems(), whereas ZenModel puts them at device.os.filesystems().

This in effect means that the proxy components have multiple containing relations (ZPL and ZenModel), which resulted the need for the additional existence test for the rel.obj before returning the actual "real" containing relname (the one that will actually work)